cpp/serve/engine_actions/action_commons.cc

```diff
@@ -29,8 +29,22 @@ void ProcessFinishedRequestStateEntries(std::vector<RequestStateEntry> finished_
     // Mark the status of this entry as finished.
     rsentry->status = RequestStateStatus::kFinished;
     // Remove the request state entry from all the models.
-    RemoveRequestFromModel(estate, rsentry->mstates[0]->internal_id, models);
-    estate->id_manager.RecycleId(rsentry->mstates[0]->internal_id);
+    if (estate->prefix_cache->HasSequence(rsentry->mstates[0]->internal_id)) {
+      if (!rsentry->request->generation_cfg->pinned) {
+        // If the request is not pinned, recycle the request.
+        estate->prefix_cache->RecycleSequence(
+            rsentry->mstates[0]->internal_id, TypedPackedFunc<void()>([estate, models, rsentry]() {
+              RemoveRequestFromModel(estate, rsentry->mstates[0]->internal_id, models);
+              estate->id_manager.RecycleId(rsentry->mstates[0]->internal_id);
+            }),
+            /*lazy=*/true);
+      }
+      // If the request is pinned, do nothing over the prefix cache and KVCache. Let the data be
```

Comment:
We should not make data orphan, instead, add a new type of state(besides active etc, make the kind is SystemKeepAlive)